src/circuit_proof/assignment.rs

```diff
@@ -137,3 +139,15 @@ impl Try for Assignment {
         Assignment::Value(v)
     }
 }
+
+impl PartialEq for Assignment {
+    fn eq(&self, other: &Assignment) -> bool {
+        match (self, other) {
+            (Assignment::Value(self_value), Assignment::Value(other_value)) => {
+                bool::from(self_value.ct_eq(other_value))
+            }
+            (Assignment::Missing(), Assignment::Missing()) => true,
+            _ => false,
+        }
+    }
+}
```

Comment:
This is implementing `PartialEq` in constant time, but because `PartialEq` returns a `bool`, it's not really effective to do the check in constant time, because the `bool` that comes out is probably going to be used in an `if` statement, which wouldn't be constant time.

Instead I think it would be better to implement the `subtle`  `ConstantTimeEq` and `ConditionallySelectable` traits for `Assignment`, and then, maybe, implement `PartialEq` in terms of those. I'm not sure if it makes sense to have a `PartialEq` implementation for `Assignment`, or whether we should try to push people to write constant-time gadgets.